pkg/kubelet/kubelet.go

```diff
@@ -472,7 +473,11 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 
 	containerRefManager := kubecontainer.NewRefManager()
 
-	oomWatcher := oomwatcher.NewWatcher(kubeDeps.Recorder)
+	oomStreamer, err := oomparser.New()
```

Comment:
If we don't want `pkg/kubelet/kubelet.go` to need to know about `cadvisor`, we could also define a `NewCadvisorWatcher` method in `pkg/kubelet/oom`, which would take care of running `oomparser.New`.

I don't have a strong preference.